core/storage/file/gae_models.py

```diff
@@ -45,34 +45,96 @@ class FileMetadataModel(base_models.VersionedModel):
 
     @classmethod
     def get_new_id(cls, entity_name):
+        """Raise an exception if an id is not created.
+
+        Args:
+            entity_name: str. An entity name that is tied to the created id.
+        """
         raise NotImplementedError
 
     @classmethod
     def get_undeleted(cls):
+        """Retrieve values that have not been deleted in the file metadata
```

Comment:
I don't understand what this means. What are "values ... in the file metadata model"?